internal/services/api/get_entrypoints.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+// Copyright (C) 2023 by Posit Software, PBC.
+
+import (
+	"encoding/json"
+	"io/fs"
+	"net/http"
+	"slices"
+
+	"github.com/posit-dev/publisher/internal/bundles/matcher"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/util"
+)
+
+func GetEntrypointsHandlerFunc(base util.AbsolutePath, log logging.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		projectDir, _, err := ProjectDirFromRequest(base, w, req, log)
+		if err != nil {
+			// Response already returned by ProjectDirFromRequest
+			return
+		}
+
+		response := []string{}
+		suffixes := []string{".htm", ".html", ".ipynb", ".py", ".qmd", ".R", ".Rmd"}
+		files := []string{
+			"*",
+			"!**/renv/activate.R",
+			"!**/_site/*.html",
+			"**/_site/index.html",
+			"!*_files",
+		}
```

Comment:
Should this skip Python virtual environments?

Reply:
Yes; that's done in the `ExcludingWalker`.